### ISSUE TITLE
Browsing meetings in details view

### ISSUE THREAD

**Issue by OSAwebsite** [code4recovery/12-step-meeting-list#53]
I use this tool for an Area with 300+ meetings. Very frequently I have to double-check information that is only in the Details screen. I would love to have an option to browse to the next or previous meeting from that Details screen.

So, in the worst case, I have to work from the full list and browse.
Sometimes it is a District list and I need to do a check on something in details view.
It is time consuming to either try the back button and choose the next one on the list, or, worse, accidentally click 'Meetings' and get back to the full list and then have to redo my selection.

It would really be fantastic if this would be added. Let me know if you need more info.

I am basically thinking of the same kind of navigation that is often seen for posts. In the bottom of the page is shows the next items and previous items. It would be fantastic to have this in the order of the list.

So, go to meetings and have all in list.
Perform a search then sort it by whatever.
Go into details.
Browse to the next one in line without having to first go back to the list, then into detail.

Possible?

Thanks,

Nicky

**Comment by lovethecodelabs** [comment]
A shortcode was put together, calling the tsml functions to loop through an array in foreach for the meetings and times and locations.  

**$mtArray = tsml_get_meetings();**  allows all meeting information to be processed by PHP within a wordpress plugin.

A simple meeting class was written and meeting objects instantiated, with a foreach loop.  Meeting information is then echoed from the returning get() method of the meeting.

Administrators have editing capabilities when viewing this table with one-click editing in the site frontend.

The code can be placed in a tsml helper plugin or perhaps merged with the existing git codebase.  Would this solve the problem?

**Comment by joshreisner** [comment]
Hey Nicky,

Technically yeah this should be possible. There's enough "state" information in the URL to infer what the previous and next meetings in the list would have been and to create links.

I'm not sure most users want / need to see this, but I agree that if you wanted to review, say, all the Wednesday Women's  meetings in your area one by one, then it would be useful for that purpose.

In the past what I've done is held down the command key (on mac, different on windows) and clicked all the links, which opens them all in separate tabs.

Not exactly sure where the prev / next links would go. Theoretically in the bottom left / bottom right corners of the screen I'm guessing?

**Comment by OSAwebsite** [comment]
Gosh, lovethecodelabs, I wish I could truly understand what you are saying. But I do not have enough coding knowledge - I think you are suggesting to create my own customised version or branch off the default code. But apart from the fact that I would not know how, it would probably leave me lost on my own island as you guys upgrade to new versions.

I am not desperate for this feature, it would just be helpful to have it. Josh, the placement of such browsing links would not really be essential. I would say top is better than bottom because you can browse and go down into detail for relevant meetings.

The feature idea came up in a bigger conversation on the Wordpress forum where somebody was wondering why I feel the need to keep an off-site database solution for my data. I feel that Wordpress/plugin is a good repository of my data, but not the most intuitive for the things I need to do with these data outside of the website. Sorting, reporting, printing, reordering... These I do daily and I cannot see myself depending on the plugin for that 100%.

Now, no bad word about the plugin: I would not have been able to design the new Area website without it. It is the number one hit of all pageviews, by far, and people intuitively know what to do with it at the front end. 

The feature is for the backend, although maybe visitors of the site would also benefit.
Wordpress is, in the end, a somewhat rigid structure for publishing data online. It was never meant to produce output the way that I need it and that is fine. I am very happy that I can import csv files from my database platform, where I do have full control over how to present the data.

But, if ever, you feel bored on a Sunday afternoon and you feel like implementing this 'previous/next' browsing based on the selected meetings in the Wordpress list, I would do a little dance of happiness!

Thanks for the feedback, much appreciated.

**Comment by lovethecodelabs** [comment]
@OSAwebsite 
Yes, we needed a plain list show all meetings, and I created a separate plugin to query the meetings contained in the Wordpress database, in the 12 Step Meeting post type format.  

I was able to get all the meetings in an array and loop through in this custom plugin.  We did not think about pagination of previous/next meeting browsing in the backend, as the entire list is editable in the Wordpress Frontend when simulataneously logged into the wordpress backend.

The plugin does not break when the tsml is updated either.

Since all meetings are displayed, I search for a meeting with Cntrl+F with any parameters needed.
